addons/account/models/account_move.py

```diff
@@ -1265,9 +1265,10 @@ def _compute_amount(self):
             total = 0.0
             total_currency = 0.0
             currencies = set()
+            lines_onchange_currency = move._get_lines_onchange_currency()
 
             for line in move.line_ids:
-                if line.currency_id and line in move._get_lines_onchange_currency():
+                if line.currency_id and line in lines_onchange_currency:
```

Comment:
Cleanup this poorly designed part of the code instead:
`currencies = move._get_lines_onchange_currency().currency_id`
Then, you can remove this from the loop

Reply:
You mean move it outside of the loop